TLS-RPT Support

jnew-gh opened the pull request:
These commits add functionality to view SMTP TLS Reports (TLS-RPT) added into the main database by dmarcts-report-parser.

Four new files, tlsts-report-viewer.php, tlsts-report-viewer-report-list.php, tlsts-report-viewer-report-data.php and json.png, are created.

jnew-gh committed:
Upgrade dmarcts- files to handle new TLS Reports

Create a new variable, report-type, to handle TLS Reports in addition to DMARC Reports

jnew-gh committed:
New tlsts- files to handle TLS Reports

These files are, in large part, a copy/paste of the dmarcts- files, with necessary changes for the different structure of the TLS Reports.

jnew-gh committed:
New JSON icon

Public domain icon json.png

ekalin commented:
Got this error when trying to view TLS reports with Postgres:
```
Uncaught PDOException: SQLSTATE[42883]: Undefined function: 7 ERROR: function year(timestamp without time zone) does not exist LINE 6: WHERE ((year(mindate) = 2022 AND month(mindate) = 7) OR (ye... ^ HINT: No function matches the given name and argument types. You might need to add explicit type casts. in /var/www/html/dmarcts-report-viewer/tlsts-report-viewer-report-list.php:268 Stack trace: #0 /var/www/html/dmarcts-report-viewer/tlsts-report-viewer-report-list.php(268): PDO->query('\nSELECT\n *\nFRO...') #1 {main} thrown in /var/www/html/dmarcts-report-viewer/tlsts-report-viewer-report-list.php on line 268
```

The problem seems to be that the `year()` function does not exist in Postgres, it must be `extract(year from mindate)` (and similarly for month), but this syntax is also valid MySQL, fortunately. See the changes I made for Postgres.

jnew-gh committed:
Make WHERE SQL clause for Periods compatible with PostgreSQL

WHERE SQL clause for Periods filter dropdown list in TLS Report viewer

jnew-gh committed:
Add whole circle css icon for TLS Report List

Add a whole circle icon in css for use in the TLS-RPT list
Reconfigure TLS Report List table to display whole circle status icon

jnew-gh committed:
Add $tls-result variable to display TLS Report result icon and text in Report List

jnew-gh committed:
Set icon colour and text in TLS Report List

jnew-gh committed:
Add TLS Report Status  to filter dropdown box

jnew-gh committed:
Implement Report Status filter for TLS Report List

jnew-gh commented:
These last 5 commits (from [f0cedef](https://github.com/techsneeze/dmarcts-report-viewer/pull/82/commits/f0cedef4f13a335c78249f2351af6937386577e5) on) add the green/yellow/red status icons to the TLS Report List and the ability to filter on the TLS Report status.

brknkfr commented:
Just tested TLS-RPT support. It seems to fork fine.

wioxjk commented:
Can we expect a merge soon?
@techsneeze
